packages/flutter/lib/src/widgets/block.dart

```diff
@@ -0,0 +1,290 @@
+
+  @override
+  _RenderSliverBlockForWidgets createRenderObject(BuildContext context) => new _RenderSliverBlockForWidgets();
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    description.add('delegate: $delegate');
+  }
+}
+
+class _SliverBlockElement extends RenderObjectElement {
+  _SliverBlockElement(SliverBlock widget) : super(widget);
+
+  @override
+  SliverBlock get widget => super.widget;
+
+  @override
+  _RenderSliverBlockForWidgets get renderObject => super.renderObject;
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    renderObject._element = this;
+  }
+
+  @override
+  void unmount() {
+    super.unmount();
+    renderObject._element = null;
+  }
+
+  @override
+  void update(SliverBlock newWidget) {
+    final SliverBlock oldWidget = widget;
+    super.update(newWidget);
+    final SliverBlockDelegate newDelegate = newWidget.delegate;
+    final SliverBlockDelegate oldDelegate = oldWidget.delegate;
+    if (newDelegate != oldDelegate &&
+        (newDelegate.runtimeType != oldDelegate.runtimeType || newDelegate.shouldRebuild(oldDelegate)))
+      performRebuild();
+  }
+
+  Map<int, Element> _childElements = new SplayTreeMap<int, Element>();
+  Map<int, Widget> _childWidgets = new HashMap<int, Widget>();
+  RenderBox _currentBeforeChild;
+  bool _debugOpenToChanges = false;
+
+  @override
+  void performRebuild() {
+    _childWidgets.clear();
+    super.performRebuild();
+    _currentBeforeChild = null;
+    assert(!_debugOpenToChanges);
+    assert(() { _debugOpenToChanges = true; return true; });
+    try {
+      // The "toList()" below is to get a copy of the array so that we can
+      // mutate _childElements within the loop. Basically we just update all the
+      // same indexes as we had before. If any of them mutate the tree, then
+      // this will also trigger a layout and so forth. (We won't call the
+      // delegate's build function multiple times, though, because we cache the
+      // delegate's results until the next time we need to rebuild the whole
+      // block widget.)
+      for (int index in _childElements.keys.toList()) {
+        Element newChild;
+        renderObject._rebuild(index, () {
+          newChild = updateChild(_childElements[index], _build(index), index);
+        });
+        if (newChild != null) {
+          _childElements[index] = newChild;
+          _currentBeforeChild = newChild.renderObject;
+        } else {
+          _childElements.remove(index);
+        }
+      }
+    } finally {
+      assert(() { _debugOpenToChanges = false; return true; });
+    }
+  }
+
+  Widget _build(int index) {
+    return _childWidgets.putIfAbsent(index, () {
+      Widget child = widget.delegate.build(this, index);
+      if (child == null)
+        return null;
+      return new RepaintBoundary.wrap(child, index);
```

Comment:
It's unclear whether we want wrap every child in a repaint boundary.  I guess we can start with that and see how it goes.